src/test/java/org/xstefank/whitelist/AddUserCommandTest.java

```diff
@@ -0,0 +1,56 @@
+package org.xstefank.whitelist;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.ArrayList;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.xstefank.model.PersistentList;
+import org.xstefank.model.Utils;
+import org.xstefank.TestUtils;
+import org.xstefank.api.GitHubAPI;
+
+import static org.powermock.api.support.membermodification.MemberMatcher.method;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({GitHubAPI.class, WhitelistProcessing.class})
+public class AddUserCommandTest {
+
+    private AddUserCommand addUserCommand;
+    private PersistentList userList = new PersistentList(TestUtils.RESOURCES_PATH, Utils.USERLIST_FILE_NAME);
+    private PersistentList adminList = new PersistentList(TestUtils.RESOURCES_PATH, Utils.ADMINLIST_FILE_NAME);
+
+    @Before
+    public void before() {
+        String regex = TestUtils.FORMAT_CONFIG.getFormat().getCommands().get(AddUserCommand.NAME);
+        addUserCommand = new AddUserCommand(regex);
+
+        PowerMockito.stub(method(GitHubAPI.class, "getJsonWithPullRequest", JsonNode.class))
+                .toReturn(TestUtils.TEST_PAYLOAD);
+
+        PowerMockito.stub(method(WhitelistProcessing.class, "loadApis"))
+                .toReturn(new ArrayList<>());
+    }
+
+    @Test
+    public void testAddUserToList() {
+       addUserCommand.process(TestUtils.ISSUE_PAYLOAD, adminList, userList);
+
+        Assert.assertTrue(userList.isUserBackedInFile("prUser"));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testProcessMethodNullParams() {
+        addUserCommand.process(null, null, null);
+    }
+
+    @After
+    public void after() {
+        TestUtils.deleteResourceFile(Utils.USERLIST_FILE_NAME);
```

Comment:
and adminlist?

Reply:
Tests reusing the adminlist everytime they are running.
  - Adminlist is always same. So it is not necessary creating it everytime the tests start.